src/Struct/Exprs.hs

```diff
@@ -19,9 +19,12 @@ data UsProg =
   | UsProgData Var [Var] [Ctor]         -- lhs, type params, constructors
   deriving (Eq, Ord)
 
+data Forall = Forall Var Bool -- name, is robust
```

Comment:
I'd mildly prefer something like `Forall Var | ForallRobust Var`...what do you think?

Reply:
or `Forall Var Bound` and `data Bound = BoundRobust | BoundNone`...anything that doesn't require remembering which one is true and which one is false...

Reply:
or let `SProgDefine` have two lists, one for unbounded type variables and one for robust-bounded type variables? Maybe @ccshan has an opinion.

Reply:
I guess "not requiring remembering which one is true and which one and false" and being able to write list comprehensions like `[ ... | Forall x r <- tpms]` argue together for `Forall Var Bound` and `data Bound = BoundRobust | BoundNone`... Not a strong opinion from me.

Reply:
OK, I think we decided to go for `Forall Var Bound` and `data Bound = BoundRobust | BoundNone` or some renaming thereof.